feature_engine/timeseries/forecasting/expanding_window_features.py

```diff
@@ -0,0 +1,232 @@
+            The dataframe with the original plus the new variables.
+        """
+        # Common dataframe checks and setting up.
+        X = super().transform(X)
+
+        tmp = (
+            X[self.variables_]
+            .expanding(min_periods=self.min_periods)
+            .agg(self.functions)
+            .shift(periods=self._infer_period(X))
+        )
+
+        tmp.columns = self.get_feature_names_out(self.variables_)
+
+        X = X.merge(tmp, left_index=True, right_index=True, how="left")
+
+        if self.drop_original:
+            X = X.drop(self.variables_, axis=1)
+
+        return X
+
+    def get_feature_names_out(self, input_features: Optional[List] = None) -> List:
+        """
+        Get output feature names for transformation.
+
+        Parameters
+        ----------
+        input_features: list, default=None
+            Input features. If `input_features` is `None`, then the names of all the
+            variables in the transformed dataset (original + new variables) is returned.
+            Alternatively, only the names for the window features derived from
+            input_features will be returned.
+
+        Returns
+        -------
+        feature_names_out: list
+            The feature names.
+        """
+        check_is_fitted(self)
+
+        # create names for all window features or just the indicated ones.
+        if input_features is None:
+            input_features_ = self.variables_
+        else:
+            if not isinstance(input_features, list):
+                raise ValueError(
+                    f"input_features must be a list. Got {input_features} instead."
+                )
+            if any([f for f in input_features if f not in self.variables_]):
+                raise ValueError(
+                    "Some features in input_features were not used as input "
+                    "for the window features. You can only get the names of "
+                    "the window features with this function."
+                )
+            # create just indicated window features
+            input_features_ = input_features
+
+        feature_names = [
+            f"{feature}_expanding_{agg}"
+            for feature in input_features_
+            for agg in self.functions
+        ]
+
+        # return names of all variables if input_features is None
+        if input_features is None:
+            if self.drop_original is True:
+                # removes names of variables to drop
+                original = [
+                    f for f in self.feature_names_in_ if f not in self.variables_
+                ]
+                feature_names = original + feature_names
+            else:
+                feature_names = self.feature_names_in_ + feature_names
+
+        return feature_names
+
+    @staticmethod
+    def _infer_period(X: pd.DataFrame | pd.Series) -> int:
+        freq = getattr(X, "inferred_freq", None)
```

Comment:
I guess I do not follow this method because I am not familiar with inferred_freq and freq_to_period.

Reply:
Will remove as shifting by periods=1 is sufficient.